internal/controller/instaslice_controller.go

```diff
@@ -407,12 +419,25 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				if podHasNodeAllocation {
 					err := utils.UpdateOrDeleteInstasliceAllocations(ctx, r.Client, instaslice.Name, allocResult, allocRequest)
 					if err != nil {
-						return ctrl.Result{Requeue: true}, nil
+						return ctrl.Result{Requeue: true}, err
 					}
 					// allocation was successful
+					// Update total processed GPU slices metrics
+					err = r.IncrementTotalProcessedGpuSliceMetrics(
```

Comment:
should we retry? InstaSlice controller may have already passed the allocation updates as seen in line 420